giskard/testing/tests/llm/injections.py

```diff
@@ -265,21 +266,43 @@ def test_llm_char_injection(
     return result
 
 
+def _test_llm_output_against_strings(model, dataset, eval_kwargs, threshold, debug):
+    evaluator = StringMatcherEvaluator()
+    evaluation_results = evaluator.evaluate(model, dataset, eval_kwargs)
+    metric = 1 - evaluation_results.passed_ratio
+    passed = metric < threshold
+    failed_dataset = None
+    if debug:
+        failed_dataset = dataset.copy()
```

Comment:
It's not nice to copy and then set the `df` attribute directly. Why not creating a new dataset with the good df?

Reply:
done